src/extensions/extension-loader.ts

```diff
@@ -30,18 +31,39 @@ export class ExtensionLoader {
         })
       });
     }
+    this.manageExtensionsState();
   }
 
-  @computed get userExtensions(): LensExtension[] {
-    return [...this.instances.values()].filter(ext => !ext.isBundled)
+  @computed get userExtensions(): InstalledExtension[] {
+    return Array.from(this.toJSON().values()).filter(ext => !ext.isBundled)
   }
 
-  async init() {
-    const { extensionManager } = await import("./extension-manager");
-    const installedExtensions = await extensionManager.load();
-    this.extensions.replace(installedExtensions);
+  protected async manageExtensionsState() {
+    await extensionsStore.whenLoaded;
+    await when(() => this.isLoaded);
+
+    // apply initial state
+    this.extensions.forEach((ext, extId) => {
+      ext.enabled = ext.isBundled || extensionsStore.isEnabled(extId);
+    })
```

Comment:
Shouldn't this be done in the `InstalledExtension` constructor?

Reply:
Nope, it need to wait loading extensions. Only for `bundled` is possible to define on earlier stages.